indexing-hadoop/src/main/java/io/druid/indexer/path/StaticPathSpec.java

```diff
@@ -96,9 +101,22 @@ public final static void addToMultipleInputs(
     // MultipleInputs.addInputPath(job, path, inputFormatClassToUse)
     // but have to handle hadoop glob path ourselves correctly
     // This change and HadoopGlobPathSplitter.java can be removed once the hadoop issue is fixed
-    for (StringBuilder sb : HadoopGlobPathSplitter.splitGlob(path)) {
-      MultipleInputs.addInputPath(job, new Path(sb.toString()), inputFormatClassToUse);
+    addInputPath(job, HadoopGlobPathSplitter.splitGlob(path), inputFormatClassToUse);
+  }
+
+  // copied from MultipleInputs.addInputPath with slight modifications
+  private static void addInputPath(Job job, Iterable<String> pathStrings, Class<? extends InputFormat> inputFormatClass)
+  {
+    Configuration conf = job.getConfiguration();
+    StringBuilder inputFormats = new StringBuilder(Strings.nullToEmpty(conf.get(MultipleInputs.DIR_FORMATS)));
+    for (String pathString : pathStrings) {
+      if (inputFormats.length() > 0) {
+        inputFormats.append(',');
+      }
+      inputFormats.append(pathString).append(';').append(inputFormatClass.getName());
     }
+    conf.set(MultipleInputs.DIR_FORMATS, inputFormats.toString());
+    job.setInputFormatClass(DelegatingInputFormat.class);
```

Comment:
why do we need to use DelegatingInputFormat, do things not work without it? just trying to minimize changes here.